### PR TITLE
[pcre2] Convert JIT to feature

### PR DESCRIPTION
**Pull request by talregev**
Convert JIT to feature

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by autoantwort** [comment]
It is currently not failing?

**Comment by talregev** [comment]
> It is currently not failing?

vcpkg ci doesn't check `arm64-osx` triplet.

**Comment by autoantwort** [comment]
I have tested it locally

**Comment by talregev** [comment]
> I have tested it locally

I tested it on osx on the gitgub action ci. and it fail without this fix.

**Comment by autoantwort** [comment]
Can you provide the error logs?

**Comment by talregev** [comment]
> Can you provide the error logs?

I will.

**Comment by autoantwort** [comment]
Ah I found it:
```
/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/pcre2/src/cre2-10.40-42fc341438.clean/src/sljit/sljitExecAllocator.c:156:2: error: "Must target Big Sur or newer"
#error "Must target Big Sur or newer"
 ^
1 error generated.
```

**Comment by autoantwort** [comment]
But then I suggest to make it a feature instead of turning it off completely because it only works on current macs.

**Comment by talregev** [comment]
> But then I suggest to make it a feature instead of turning it off completely because it only works on current macs.

I will try to convert it to a feature.

File: ports/pcre2/vcpkg.json [hunk 1]
```diff
@@ -19,5 +19,14 @@
...
     "zlib"
-  ]
+  ],
+  "default-features": [
+    "jit"
+  ],
+  "features": {
+    "jit": {
+      "description": "Build JIT support",
+      "supports": "!ios & !emscripten"
```
Comment (autoantwort) [inline ports/pcre2/vcpkg.json:29]: The `!ios` is actually wrong. I tested it locally.

Comment (talregev) [reply ports/pcre2/vcpkg.json:29]: Did you see that in the original they turn it off for ios?

Comment (autoantwort) [reply ports/pcre2/vcpkg.json:29]: Yes, but it seems that this is wrong (maybe it was right in the past)

Comment (talregev) [reply ports/pcre2/vcpkg.json]: I remove it. But I have a feeling that other people will fix it to add not support ios.

**Comment by talregev** [comment]
@autoantwort Can you test my port and check I didn't make errors?

File: ports/pcre2/vcpkg.json (hunk 1 above)
Comment (autoantwort) [reply ports/pcre2/vcpkg.json]: We'll see in the future.

**Review by autoantwort — DISMISSED** [review]

**Comment by talregev** [comment]
@autoantwort What about android failing?

**Comment by autoantwort** [comment]
I think it is a baseline regression not caused by this PR.

**Comment by Osyotr** [comment]
It think that until https://github.com/microsoft/vcpkg-tool/pull/813 is merged, `default-features` should not contain features that have `supports` expression.
Workaround is to create `platform-default-features` feature that has `platform` expression (https://github.com/microsoft/vcpkg/pull/31291/files#diff-fa5764c3245e6fdc2a5bd9ad48405b4e6f2555a4bff9079d365bb4990e3b002a)

**Review by Cheney-W — DISMISSED** [review]

**Comment by talregev** [comment]
@Cheney-W 
There already fix that approved
https://github.com/microsoft/vcpkg/pull/29767

I will merge the master after you will merge it.

**Comment by dg0yt** [comment]
> It think that until [microsoft/vcpkg-tool#813](https://github.com/microsoft/vcpkg-tool/pull/813) is merged, `default-features` should not contain features that have `supports` expression. Workaround is to create `platform-default-features` feature that has `platform` expression (https://github.com/microsoft/vcpkg/pull/31291/files#diff-fa5764c3245e6fdc2a5bd9ad48405b4e6f2555a4bff9079d365bb4990e3b002a)

This still must be done.

File: ports/pcre2/vcpkg.json [hunk 2]
```diff
@@ -19,5 +19,14 @@
...
+  ],
+  "default-features": [
+    "jit"
+  ],
+  "features": {
+    "jit": {
+      "description": "Build JIT support",
+      "supports": "!emscripten"
+    }
+  }
```
Comment (vicroms) [inline ports/pcre2/vcpkg.json:31]:
```suggestion
  "default-features": [
    "default-features"
  ],
  "features": {
    "default-features": {
      "description": "Enable default features",
      "dependencies": [
        {
          "name": "pcre2",
          "platform": "!emscripten",
          "features": [
            "jit"
          ]
        }
      ]
    },
    "jit": {
      "description": "Build JIT support",
      "supports": "!emscripten"
    }
  }
```

Comment (vicroms) [reply ports/pcre2/vcpkg.json:31]: As suggested by @Osyotr make a `default-features` a feature itself

**Review by vicroms — DISMISSED** [review]

**Comment by Cheney-W** [comment]
I have converted your PR to draft status. When you respond to Victor's suggestion, please revert to "ready for review". That way, I can be aware that you've responded since you can't modify the tags.

File: ports/pcre2/vcpkg.json (hunk 2 above)
Comment (talregev) [reply ports/pcre2/vcpkg.json]: done.

**Comment by talregev** [comment]
@Cheney-W @vicroms 
Ready for review.

**Review by Cheney-W — APPROVED** [review]

**Pull request merged**